### DISCUSSION TITLE
How do I change the notification icon?

### DISCUSSION THREAD

**Discussion by parksb** [tauri-apps/tauri#7686] [Q&A] [answered]
Hi, I'm struggling with a notification icon on macOS. I have my own icon files in `src-tauri/icons` directory. The program icon is set to my icon, but the icon in the notification is not.

I wrote a simple back-end that sends a notification every second for example:

```rust
fn main() {
  tauri::Builder::default()
    .setup(|app| {
      let app_id = app.config().tauri.bundle.identifier.clone();

      thread::spawn(move || loop {
        Notification::new(&app_id)
          .title("Title")
          .body("Body")
          .show();
        thread::sleep(Duration::from_secs(1));
      });

      Ok(())
    })
    .run(tauri::generate_context!());
}
```

I expected the notification would be shown with the icon I set after running `pnpm tauri dev`, but it was shown with Terminal icon.

![image](https://github.com/tauri-apps/tauri/assets/6410412/a4ef7ea8-b5f6-45b2-8a65-e1e355210208)

When I ran the program built by `pnpm tauri build`, a notification was shown with Tauri icon.

![image](https://github.com/tauri-apps/tauri/assets/6410412/29e319eb-9ccc-4e23-9b2d-8903a930f972)

The following screenshot shows icon files generated by `pnpm tauri icon`:

![image](https://github.com/tauri-apps/tauri/assets/6410412/10f1e950-c03a-4910-a0bd-66632336bdc9)

How do I set the notification icon to I made?

**Comment by FabianLars** [comment, marked as answer]
I apologize that this is _still_ not documented in our own docs, but only in the upstream crate's docs: https://docs.rs/notify-rust/latest/notify_rust/struct.Notification.html#method.icon
tldr: On macOS you can't use random-ish icons and only those of installed applications, which is also why we use the terminal icon in dev because at that point your app isn't recognized as an actual app.

**Reply by parksb** [reply]
Thank you for answering :) So, is it an issue that needs to be solved on notify-rust? Or is it an inevitable problem?

**Reply by FabianLars** [reply]
It's is a macos limitation. A semi-common workaround other apps/frameworks is to add a second icon on the right side of the notification which can be modified, but that may be out of scope of notiy-rust. (Also as a new part-time macos user i think this looks _really_ weird)
